src/mappings/mappingHandlers.ts

```diff
@@ -1,277 +1,99 @@
-      datetime: block.timestamp,
-      specVersionId: block.specVersion,
-    }).save();
+      countExtrinsicsUnsigned,
+    });
+
+    await Promise.all([
+      store.bulkCreate('Extrinsic', extrinsics),
+      store.bulkCreate('Event', events),
+      block.save(),
+    ]);
+    logger.debug(`Processed block - ${block.id} for all events and extrinsics`);
   } catch (error) {
-    logError(`Received an error in handleBlock: ${error.toString()}`);
+    logError(`Received an error while processing block ${blockId}: ${error.toString()}`);
     throw error;
   }
 }
 
-const processBlockExtrinsics = (extrinsics: Vec<GenericExtrinsic<AnyTuple>>) => {
-  const ret = {
-    countExtrinsicsUnsigned: 0,
-    countExtrinsicsSigned: 0,
-  };
-  for (const extrinsic of extrinsics) {
-    if (extrinsic.isSigned) {
-      ret.countExtrinsicsSigned++;
+async function getEvents(
+  substrateBlock: SubstrateBlock,
+  wrappedExtrinsics
+): Promise<{
+  events: Event[];
+  countExtrinsicsSuccess: number;
+}> {
+  let countExtrinsicsSuccess = 0;
+  const events: Event[] = [];
+  for (const [i, event] of substrateBlock.events.entries()) {
+    if (
+      event.event.section.toLowerCase() === ModuleIdEnum.system &&
+      event.event.method === EventIdEnum.ExtrinsicSuccess
+    ) {
+      countExtrinsicsSuccess++;
     } else {
-      ret.countExtrinsicsUnsigned++;
+      let extrinsic: SubstrateExtrinsic;
+      if (event.phase.isApplyExtrinsic) {
+        extrinsic = wrappedExtrinsics.find(({ idx }) => event.phase.asApplyExtrinsic.eqn(idx));
+      }
+      const dbEvent = await handleEvent(event as EventRecord, i, substrateBlock, extrinsic);
+      events.push(dbEvent);
     }
   }
-  return ret;
-};
-
-export async function handleToolingEvent(event: SubstrateEvent): Promise<void> {
-  const genericEvent = event.event as GenericEvent;
-  const block = event.block;
-  const blockId = block.block.header.number.toString();
-  const eventIdx = event.idx;
-  const moduleId = genericEvent.section.toLowerCase();
-  const eventId = genericEvent.method;
-  const args = genericEvent.data;
+  return {
+    events,
+    countExtrinsicsSuccess,
+  };
+}
 
-  const harvesterLikeArgs = args.map((arg, i) => {
-    let type;
-    const typeName = genericEvent.meta.fields[i].typeName;
-    if (typeName.isSome) {
-      // for metadata >= v14
-      type = typeName.unwrap().toString();
+function getExtrinsics(wrappedExtrinsics): {
+  extrinsics: Extrinsic[];
+  countExtrinsicsSigned: number;
+  countExtrinsicsUnsigned: number;
+} {
+  let countExtrinsicsUnsigned = 0;
+  let countExtrinsicsSigned = 0;
+  const extrinsics: Extrinsic[] = [];
+  for (const ext of wrappedExtrinsics) {
+    if (ext.extrinsic.isSigned) {
+      countExtrinsicsSigned++;
     } else {
-      // for metadata < v14
-      type = genericEvent.meta.args[i].toString();
+      countExtrinsicsUnsigned++;
+    }
+    if (
+      ext.extrinsic.method.section.toLowerCase() !== ModuleIdEnum.timestamp ||
```

Comment:
I was thinking we would use mapping filters in the manifest, but we'd need to switch over to event or call level handlers.

I am not sure how much work this would be. I would think the events have reference to the block.

https://academy.subquery.network/build/manifest/polkadot.html#mapping-handlers-and-filters